docs/source/cuda._sanitizer.rst

```diff
@@ -0,0 +1,95 @@
+.. currentmodule:: torch.cuda._sanitizer
+
+torch.cuda._sanitizer
+=====================
+
+Overview
+--------
+
+.. automodule:: torch.cuda._sanitizer
+
+
+Usage
+------
+
+Here is an example of a simple synchronization error in PyTorch:
+
+::
+
+    import torch
+
+    a = torch.rand(4, 2, device="cuda")
+
+    with torch.cuda.stream(torch.cuda.Stream()):
+        torch.mul(a, 5, out=a)
+
+The ``a`` tensor is initialized on the default stream and, without any synchronization
+methods, modified on a new stream. When this script is run on the commandline with:
+::
+
+    TORCH_CUDA_SANITIZER=1 python example_error.py
+
+the following output is printed by CSAN:
+
+::
+
+    ============================
+    CSAN detected a possible data race on tensor with data pointer 139719969079296
+    Access by stream 94646435460352 during kernel:
+    aten::mul.out(Tensor self, Tensor other, *, Tensor(a!) out) -> Tensor(a!)
+    writing to argument: self, out, output
```

Comment:
[nit] I'm just seeing it now: it says "writing to argument output", but there is no argument called output. I guess it means that the problematic tensor is the output of the operator. Can it be made clearer? Something like "writing to arguments self and out, and to the output"?

Reply:
I have a future rework in mind that will address this issue, but I will add this as a separate PR.